### PR TITLE
BE-4632: add prepared error from response

### PR DESCRIPTION
**Pull request by ivanu74**

### CODE REVIEW

File: es.go [hunk 1]
```diff
@@ -10,6 +10,8 @@ import (
 	"github.com/elastic/go-elasticsearch/v7"
 )
 
+type Client = elasticsearch.Client
```
Comment (mkorolyov) [inline es.go]: what for?

File: error.go [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+func (e Error) Error() string {
+	return fmt.Sprintf("Error: [%d] %s: %s",
+		e.StatusCode,
+		e.Type,
+		e.Reason,
+	)
+}
+
+func ExtractError(res *esapi.Response) error {
+	if res.IsError() {
```
Comment (mkorolyov) [inline error.go]: lets reduce nesting? we can return early here and reduce nesting `if !res.IsError() { return nil }`

**Review by mkorolyov — CHANGES_REQUESTED** [review]

File: error.go (hunk 2 above)
Comment (ivanu74) [reply error.go]: ok

File: es.go (hunk 1 above)
Comment (ivanu74) [reply es.go]: it's only boilerplate to avoid import. I'll remove it.

**Pull request merged**